liblwgeom/lwgeom_geos.c

```diff
@@ -829,6 +829,55 @@ lwgeom_centroid(const LWGEOM* geom)
 	return result;
 }
 
+LWGEOM *
+lwgeom_pointonsurface(const LWGEOM *geom)
+{
+	LWGEOM *result;
+	int32_t srid = get_result_srid(geom, NULL, __func__);
+	uint8_t is3d = FLAGS_GET_Z(geom->flags);
+	GEOSGeometry *g1, *g3;
+
+	if (srid == SRID_INVALID) return NULL;
+
+	if (lwgeom_is_empty(geom))
+	{
+		LWPOINT *lwp = lwpoint_construct_empty(srid, is3d, lwgeom_has_m(geom));
+		return lwpoint_as_lwgeom(lwp);
+	}
+
+	initGEOS(lwnotice, lwgeom_geos_error);
+
+	if (!input_lwgeom_to_geos(&g1, geom, __func__)) return NULL;
+
+	g3 = GEOSPointOnSurface(g1);
+
+	if (!g3)
+	{
+		GEOSGeometry *g1v;
+		lwnotice("%s: GEOS Error: %s", __func__, lwgeom_geos_errmsg);
+
+		if (!GEOSisValid(g1))
```

Comment:
It would really be good to complete the discussion on this issue with the project team before continuing to make changes like this.

Reply:
Concur, we have long shied away from automagically fixing people's geometry.